src/rgw/services/svc_sys_obj_core.cc

```diff
@@ -67,9 +67,11 @@ int RGWSI_SysObj_Core::get_system_obj_state_impl(RGWSysObjectCtxBase *rctx,
   }
 
   s->obj = obj;
-
-  int r = raw_stat(obj, &s->size, &s->mtime, &s->epoch, &s->attrset,
-                   (s->prefetch_data ? &s->data : nullptr), objv_tracker, y);
+  int r = -ENOENT;
+  if(obj.oid.find(RGW_USER_ANON_ID) == std::string::npos) {
```

Comment:
the system object service shouldn't know/care about specific object names. we need to catch these calls at a higher level - for example, one comes from:
```
#0  RGWUserCtl::get_info_by_uid
#1  RGWUserCtl::get_attrs_by_uid
#2  rgw_build_bucket_policies
#3  RGWHandler::do_init_permissions
#4  RGWHandler_REST::init_permissions
#5  rgw_process_authenticated
```

Reply:
@cbodley  can you take a look?